mm/ktsan/func.c

```diff
@@ -1,5 +1,31 @@
 #include "ktsan.h"
 
+#include <linux/preempt.h>
+
+static void kt_tame_acquire_and_release_clocks(kt_thr_t *thr)
+{
+	if (in_interrupt())
+		return;
+
+	if (thr->acquire_active) {
+		thr->acquire_tame_counter++;
+		if (thr->acquire_tame_counter == KT_TAME_COUNTER_LIMIT) {
+			thr->acquire_active = false;
+			thr->acquire_tame_counter = 0;
```

Comment:
it will be less code if you do:
- remove acquire_tame_counter
- make acquire_active an unsigned variable
- when enabling barriers, set acquire_active = KT_TAME_COUNTER_LIMIT
- here do: if (thr->acquire_active) thr->acquire_active--
- when you need to check whether barrier is enabled or not: if (thr->acquire_active)